pkg/agent/core/ngt/service/kvs/kvs_test.go

```diff
@@ -121,62 +123,140 @@ func Test_bidi_Get(t *testing.T) {
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           key: "",
-		       },
-		       fields: fields {
-		           ou: nil,
-		           uo: nil,
-		           l: 0,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           key: "",
-		           },
-		           fields: fields {
-		           ou: nil,
-		           uo: nil,
-		           l: 0,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+
+			return test{
+				name: "return the value when there is a value for the key",
+				args: args{
+					key: key,
+				},
+				fields: fields,
+				beforeFunc: func(a args, bm BidiMap) {
+					bm.Set(a.key, val)
+				},
+				want: want{
+					want:  val,
+					want1: true,
+				},
+			}
+		}(),
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key1        = "45637ec4-c85f-11ea-87d0"
+				val1 uint32 = 14438
+			)
+			var key2 = "84a333-59633fd4-4553-414a"
+
+			return test{
+				name: "return (0, false) when there is no value for the key",
+				args: args{
+					key: key2,
+				},
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key1, val1)
+				},
+				want: want{
+					want:  0,
+					want1: false,
+				},
+			}
+		}(),
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+
+			return test{
+				name:   "return (0, false) when there is no value for the key and the key is empty string",
+				args:   args{},
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key, val)
+				},
+				want: want{
+					want:  0,
+					want1: false,
+				},
+			}
+		}(),
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = ""
+				val uint32 = 0
+			)
+
+			return test{
+				name:   "return (0, false) when default value is set and there is value for the key and the key is default value",
+				args:   args{},
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key, val)
+				},
+				want: want{
+					want:  0,
+					want1: true,
+				},
+			}
+		}(),
```

Comment:
sorry again but can you add new test cases when the same key is set twice and see if it will override the value that is set?

Reply:
thank you 
I fixed: https://github.com/vdaas/vald/pull/1223/commits/057528c256e881565370bccf7074687264ecb9b4

Reply:
Why don't you test when `l != 0`, `l = math.MaxUint64()`?